Generate unique manifest file name when sideloading on Mac

TCourtneyOwen opened the pull request:
- Changes ensure we always generate a unique manifest file name to cache in the WEF folder on Mac during sideloading
- Remove check for duplicate manifest in WEF. If we encounter a duplicate manifest, just overwrite it
- Update manifest remove code to construct the manifest file path for removal using the manifest guid

TCourtneyOwen committed:
Generate unique manifest file name when sideloading on Mac

- Cache unique manifest file name in WEF cache on Mac to mitigate issues where we've already cached a manifest with the same name. By using the manifest guid in the file name, we should always have unique manifest file names in the WEF cache
- Remove check for duplicate instance of manifest in the WEF cache. If there is a duplicate instance of the manifest found in the WEF cache during sideloading, now we will simply overwrite it
- Update remove code to use the manifest guid to contruct the manifest file path that's to be removed

TCourtneyOwen committed:
Accidentally neglected to include change in previous commit

akrantz commented on src/util.ts:
```diff
@@ -131,20 +131,12 @@ function removeManifest(application: string, manifestPath: string): Promise<any>
 // NON-WIN32 COMMANDS //
 function addManifestToSideloadingDirectory(application: string, manifestPath: string): Promise<any> {
-  return new Promise ((resolve, reject) => {
+  return new Promise (async (resolve, reject) => {
```
I don't think it is a good idea to mix code that returns new Promise() with an async function. 

I don't even see any code that needs to declare the promise. Instead, I believe you can make it `async function addManifestToSideloadingDirectory` and removing the `return new Promise`. When the function completes, it will then resolve. 

This should also be Promise<void> as I only see `resolve()` in the code now and no actual value returned via the promise.

akrantz commented on src/util.ts:
```diff
@@ -131,20 +131,12 @@ function removeManifest(application: string, manifestPath: string): Promise<any>
-      }
-    }
+    const [type, manifestGuid, version] = await parseManifest(manifestPath);
+    const sideloadingManifestPath = path.join(sideloadingDirectory, manifestGuid.concat(".", path.basename(manifestPath)));
```
nit: use `path.join(sideloadingDirectory, ${manifestGuid}.${path.basename(manifestPath)})`

TCourtneyOwen replied:
fixed

akrantz commented on src/util.ts:
```diff
@@ -180,17 +172,18 @@ function removeManifestFromSideloadingDirectory(inputApplication: string, manife
-            if (sideloadingManifestPath === realManifestPath) {
-              console.log(`Removing ${sideloadingManifestPath} for application ${application}`);
-              fs.unlinkSync(sideloadingManifestPath);
+            const sideloadedManifestPath = path.join(sideloadingDirectory, manifestGuid.concat(".", path.basename(manifestPathToRemove)));
```
Same comment as before. Since the same code is in two places, perhaps move it into a single function rather than have duplicate code.

TCourtneyOwen replied:
created getSideloadManifestPath method

TCourtneyOwen committed:
Address comments from PR; create getSideloadManifestPath method

- Note, I am still trying to figure out why we are getting a promise warning but this is unrelated to my changes and an issue that's existed for some time now

TCourtneyOwen committed:
Add try/catch to addManifestToSideloadingDirectory

-Remove new Promise from removeManifestFromSideloadingDirectory to address unhandled promise warning

akrantz commented on src/util.ts:
```diff
@@ -131,11 +131,16 @@ function removeManifest(application: string, manifestPath: string): Promise<any>
+    return fs.ensureLinkSync(manifestPath, sideloadingManifestPath);
+
+  } catch (err) {
+    throw new Error(`addManifestToSideloadingDirectory failed with error:\n${err}`);
```
Suggestion for error string: `Unable to add the manifest to the sideloading directory.\n${err}`

akrantz commented on src/util.ts:
```diff
@@ -159,38 +164,35 @@ function getManifestsFromSideloadingDirectory(inputApplication: string): Promise
-  });
+    return console.log('No manifests were found to remove. Use "list" to show manifests that have been added.');
+  } catch (err) {
+    throw new Error(`removeManifestFromSideloadingDirectory failed with error:\n${err}`);
```
Suggest: `Unable to remove the manifest from the sideloading directory.\n${err}`

TCourtneyOwen committed:
Update error messages per suggestion

akrantz commented on src/util.ts:
```diff
@@ -139,7 +139,7 @@ async function addManifestToSideloadingDirectory(application: string, manifestPa
 
   } catch (err) {
-    throw new Error(`addManifestToSideloadingDirectory failed with error:\n${err}`);
+    throw new Error(`Unable to add manifest to the sideloading directory\n${err}`);
```
nit: Can you please add "the" and a period":  Unable to add **the** manifest to the sideloading directory.

The other message already has it.

TCourtneyOwen committed:
Add missing period in error message

TCourtneyOwen commented:
I believe all outstanding comments have been addressed
